### PR TITLE
Create unit tests for ParserUtil#parseX() #401

### PR DESCRIPTION
**Pull request by limmlingg**
Part of #401

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@limmlingg submitted v1 for review.

* [v1 1/1] [Add unit tests for ParserUtil\#parseX\(\) ](https://github.com/se-edu/addressbook-level4/pull/425/commits/7a24a37c5af46664d30a0e5d45f9225a51b4685c)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/ef06553753b5073c012ccac9ab1082c46b9d0f51...7a24a37c5af46664d30a0e5d45f9225a51b4685c))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/425/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    @Test
+    public void parseName_optionalEmpty_returnsOptionalEmpty() throws Exception {
+        assertFalse(ParserUtil.parseName(Optional.empty()).isPresent());
+    }
+
+    @Test
+    public void parseName_validValue_returnsName() throws Exception {
+        Name expectedName = new Name("Name 123");
+        Optional<Name> actualName = ParserUtil.parseName(Optional.of("Name 123"));
```
Comment (damithc) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java]: `"Name 123"` appears twice. Extract as a local variable? A few similar cases in the code below.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    }
+
+    @Test
+    public void parseEmail_optionalEmpty_returnsOptionalEmpty() throws Exception {
+        assertFalse(ParserUtil.parseEmail(Optional.empty()).isPresent());
+    }
+
+    @Test
+    public void parseEmail_validValue_returnsEmail() throws Exception {
+        Email expectedEmail = new Email("example@email.com");
```
Comment (damithc) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java]: I think we use `example.com` for dummy email addresses. It's a common practice. see [example.com](example.com)

**Comment by CanIHasReview** [comment]
# v2

@limmlingg submitted v2 for review.

* [v2 1/1] [Add unit tests for ParserUtil\#parseX\(\) ](https://github.com/se-edu/addressbook-level4/pull/425/commits/d0d6fae97802d4694b75c8c5f0e477607435c034)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/ef06553753b5073c012ccac9ab1082c46b9d0f51...d0d6fae97802d4694b75c8c5f0e477607435c034)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/3951074e273ff6a9f7a0c791f6b80ab9ebfafde2~1...3951074e273ff6a9f7a0c791f6b80ab9ebfafde2))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/425/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Pull request merged**